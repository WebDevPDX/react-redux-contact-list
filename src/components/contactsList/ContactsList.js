import React, { Component }from 'react'
import { AddButton } from '../addButton/AddButton'
import { DetailsButton } from'../../components/detailsButton/DetailsButton'
import { DeleteButton } from '../../components/deleteButton/DeleteButton'

class ContactsList extends Component {

  deleteHandler(contact) {
    this.props.deleteContact(contact)
  }

  createContactLine(contact) {
    return(
      <div className='box' key={contact._id}>
      <div className='columns'>
        <div className='column'>
          <div className='columns is-mobile'>
            <div className='column'>{contact.lastName}</div>
            <div className='column'>{contact.firstName}</div>
          </div>
        </div>
        <div className='column'>
          <div className='columns is-mobile'>
            <div className='column'>
              <DetailsButton contact={contact}/>
            </div>
            <div className='column'>
              <DeleteButton deleteContact={() => this.deleteHandler(contact)}/>
            </div>
          </div>
        </div>
      </div>
      </div>
    )
  }

  render () {
    return (
      <div className=''>
        <div className='column'>
          <AddButton />
        </div>
        <div className='column'>
          {this.props.contacts.map(contact => this.createContactLine(contact))}
        </div>
      </div>
    )
  }  
}

export default ContactsList