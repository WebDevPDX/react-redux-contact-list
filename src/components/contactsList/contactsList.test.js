import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import ContactsList from './ContactsList'

it('renders correctly', () => {
  const renderer = new ShallowRenderer()
  const mockContacts = [
    {
      _id: 'testID123',
      firstName: 'ExampleFN',
      lastName: 'ExampleLN',
      email: 'fn.ln@gmail.com',
      country: 'US',
    },
  ]
  const tree = renderer
    .render(<ContactsList contacts={mockContacts} />)

  expect(tree).toMatchSnapshot();
});