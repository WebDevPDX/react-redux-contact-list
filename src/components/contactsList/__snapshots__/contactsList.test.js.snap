// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className=""
>
  <div
    className="column"
  >
    <AddButton />
  </div>
  <div
    className="column"
  >
    <div
      className="box"
    >
      <div
        className="columns"
      >
        <div
          className="column"
        >
          <div
            className="columns is-mobile"
          >
            <div
              className="column"
            >
              ExampleLN
            </div>
            <div
              className="column"
            >
              ExampleFN
            </div>
          </div>
        </div>
        <div
          className="column"
        >
          <div
            className="columns is-mobile"
          >
            <div
              className="column"
            >
              <DetailsButton
                contact={
                  Object {
                    "_id": "testID123",
                    "country": "US",
                    "email": "fn.ln@gmail.com",
                    "firstName": "ExampleFN",
                    "lastName": "ExampleLN",
                  }
                }
              />
            </div>
            <div
              className="column"
            >
              <DeleteButton
                deleteContact={[Function]}
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
